src/juju/jimm-facade.ts

```diff
@@ -3,10 +3,24 @@ import { autoBind } from "@canonical/jujulib/dist/api/utils";
 
 import type { Controller } from "store/juju/types";
 
+// As typed in JIMM:
+// https://github.com/canonical/jimm/blob/1da76ed2d8dba741f5880f32c073f85f7d518904/api/params/params.go#L344
+export type CrossModelQueryRequest = {
+  type: string;
+  query: string;
+};
+
+// As typed in JIMM:
+// https://github.com/canonical/jimm/blob/1da76ed2d8dba741f5880f32c073f85f7d518904/api/params/params.go#L353
+export type CrossModelQueryResponse = {
+  results: Record<string, any>;
+  errors: Record<string, string>;
+};
```

Comment:
It appears that if you pass an invalid type or query the response is just a string. So that you can refer to the `errors` and `results` types in store/juju/types.ts it might be good to keep this object type separate under a different name e.g.:

```suggestion
export type CrossModelQueryFullResponse = {
  results: Record<string, unknown[]>;
  errors: Record<string, string[]>;
};

export type CrossModelQueryResponse = CrossModelQueryFullResponse | string;
```